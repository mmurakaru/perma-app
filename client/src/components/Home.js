import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import { v4 as uuidv4 } from 'uuid'
import { ReactComponent as Illustration } from '../assets/illustration.svg'

Home.propTypes = {
    space: PropTypes.object,
}

export default function Home({ space }) {
    return (
        <>
            <IllustrationStyled />
            <PlantsWrapper>
                {space !== undefined && <h2>Recently planted</h2>}
                {space !== undefined &&
                    space.plants.map((plant) => (
                        <Plant key={uuidv4()}>
                            <PlantImage src={plant.image_url} alt="" />
                            <PlantInfo>
                                <h2>{plant.common_name}</h2>
                                <span>{space.date}</span>
                            </PlantInfo>
                        </Plant>
                    ))}
            </PlantsWrapper>
        </>
    )
}

const IllustrationStyled = styled(Illustration)`
    width: 180px;
    position: absolute;
    top: 0px;
    right: 0px;
`

const PlantsWrapper = styled.div`
    margin-top: 70%;
    overflow: auto;
    height: 50vh;
`

const Plant = styled.div`
    border: 0.2px solid var(--medium-grey);
    border-radius: 6px;
    padding: 5px 0 5px;
    display: flex;
    align-items: center;
    justify-content: flex-start;
    margin-top: 5px;
`

const PlantInfo = styled.div`
    margin-left: 20px;
    display: grid;
    grid-template: column;
    gap: 5px;

    h2 {
        font-weight: 400;
        font-size: 0.8rem;
    }

    span {
        font-size: 0.6rem;
        font-weight: 200;
    }
`

const PlantImage = styled.img`
    width: 40px;
    height: 40px;
    border-radius: 50%;
    margin-left: 10px;
`
