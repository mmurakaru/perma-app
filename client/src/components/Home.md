Home example:

```js
<Home
    space={{
        id: 1234,
        spaceTitle: 'spaceTitle',
        plants: [
            {
                common_name: 'Pine',
                image_url:
                    'https://bs.floristic.org/image/o/63073d2fbf45b90701279405ecc2eec0272906ed',
                family: 'Trees',
            },
            {
                common_name: 'Strawberry',
                image_url:
                    'https://bs.floristic.org/image/o/63073d2fbf45b90701279405ecc2eec0272906ed',
                family: 'Trees',
            },
        ],
        date: '20.12.2020',
    }}
/>
```
