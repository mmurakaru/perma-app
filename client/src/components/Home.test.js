import Home from './Home'
import { render } from '@testing-library/react'

describe('Home', () => {
    it('renders correctly', () => {
        const { container } = render(
            <Home
                space={{
                    id: 1234,
                    spaceTitle: 'spaceTitle',
                    plants: [
                        {
                            common_name: 'Pine',
                            image_url: 'https',
                            family: 'Trees',
                        },
                    ],
                    date: '20.12.2020',
                }}
            />
        )
        expect(container.firstChild).toMatchSnapshot()
    })
})
