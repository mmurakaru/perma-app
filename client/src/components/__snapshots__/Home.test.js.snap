// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
.c0 {
  width: 180px;
  position: absolute;
  top: 0px;
  right: 0px;
}

<svg
  class="c0"
>
  illustration.svg
</svg>
`;
